### ISSUE TITLE
Sign OPT attack on Imagenet

### ISSUE THREAD

**Issue by maithal** [cmhcbb/attackbox#11]
Hi,

I tried using the Sign-OPT attack on 1 train image of Imagenet. For some reason the attack doesn't occur and the code stops running.

Loading ImageNet validation dataset.
Image batch: 0
Targeted attack - Source: 1 and Target: 4 Seed: 10
Searching for the initial direction on 100 samples: 
05/05 01:53:09 AM ==========
05/05 01:53:09 AM clean count:1
05/05 01:53:09 AM acc under attack count:1

It doesn't even find the initial distortion. Is there any specific reason why it is doing this?

Any help will be highly appreciated!

**Comment by cmhcbb** [comment]
Hi,
   It is possible that we couldn't find an initial distortion because it uses the normal distribution as its initial. You could try to initiate from any image with a different class from the dataset. You only need to change the initial code.

**Comment by maithal** [comment]
Thanks for your quick response!

When I change the input image, the true label of image is not provided and hence I get following issue:

05/06 03:25:22 PM image batch: 0
Fail to classify the image. No need to attack.

Is there any reason why the test_loader fail to provide correct labels?

Please advise!

**Comment by cmhcbb** [comment]
I think you feed the attacker with a wrong classified example. If the example is not correctly classified, there is no need to attack. What I meant previously is the initialization should start from another example with a different label.

**Comment by maithal** [comment]
I really appreciate your quick response & help!

However, can you explain how the ImageFolder is passing right labels for your original code (referred from jupyter notebook file in sign-sgd folder) but when I tried running it didn't pass equivalent labels?

Any help will be highly appreciated!

Thanks again!

**Comment by cmhcbb** [comment]
Please check https://pytorch.org/vision/master/datasets.html#imagenet for details. Pytorch has provided the loader and it should provide you with the correct label. To be noted, if it is from the validation set, you have to process the validation files first. Check this line https://gist.github.com/BIGBALLON/8a71d225eff18d88e469e6ea9b39cef4#file-extract_ilsvrc-sh-L34.

**Comment by maithal** [comment]
Thank you sharing the information! It works now.

Really appreciate your help.
